File: java/client/src/main/java/glide/api/models/configuration/CompressionBackend.java
/** Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0 */
package glide.api.models.configuration;

/**
 * Represents the compression backend to use for automatic compression.
 *
 * @see CompressionConfiguration
 */
public enum CompressionBackend {
    /** Use zstd compression backend. Default compression level is 3. */
    ZSTD,
    /** Use lz4 compression backend. Default compression level is 0. */
    LZ4
}

File: csharp/lib/CompressionBackend.cs
// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0

namespace Glide;

/// <summary>
/// Represents the compression backend to use for automatic compression.
/// </summary>
public enum CompressionBackend
{
    /// <summary>
    /// Use zstd compression backend. Default compression level is 3.
    /// </summary>
    Zstd = 0,

    /// <summary>
    /// Use lz4 compression backend. Default compression level is 0.
    /// </summary>
    Lz4 = 1,
}
